libvast/src/system/archive.cpp

```diff
@@ -185,9 +185,17 @@ archive(archive_type::stateful_pointer<archive_state> self, path dir,
           t.stop(events);
         },
         [=](unit_t&, const error& err) {
-          if (err) {
-            VAST_ERROR(self, "got a stream error:", self->system().render(err));
+          // We get an 'unreachable' error when the stream becomes unreachable
+          // because the actor was destroyed; in this case we can't use `self`
+          // anymore.
+          if (err && err != caf::exit_reason::unreachable) {
+            if (err != caf::exit_reason::user_shutdown)
+              VAST_ERROR(self, "got a stream error:", render(err));
```

Comment:
I think we should still print the error on `DEBUG`, to be able to distinguish between unreachable and user_shutdown in the logs.